### DISCUSSION TITLE
Task got removed without obvious reason

### DISCUSSION THREAD

**Discussion by MaksymSofer** [apache/airflow#18744] [Q&A]
Hello,

We are running 2.0.2 and some time ago we updated one of our DAG adding new task (based on  BashOperator).
Now come time to use it - but when we are trying to execute it - the new task is present only for a few minutes and than marked as " removed" - in DAG details stated the new task is present in removed state
The dag was updated by pushing changes to git and than (once file was delivered to airflow machine) hitting "refresh" button in UI.

I assume that i`m missing some point here - please help.

**Comment by potiuk** [comment]
Check if you have no second scheduler runnig with old version of the DAG file. This MIGHT happen that some old stale processes continue to run. Or maybe you have some process that brings the old version of DAG from somewhere else (maybe another file, stale .pyc file, etc.)

**Reply by MaksymSofer** [reply]
Hi, thank you for response:
We have only one scheduler running.
I tried to restart our whole kubernetes cluster and once restarted - he fetched updated DAG, but the next change once again is not being processed properly.
I checked adding new DAG (the same old one with changed name of the file and in code) it was properly read.

Does the data cached somewhere, like in DB ?

**Reply by MaksymSofer** [reply]
Our issue seems to be very similar to one discussed here:
https://stackoverflow.com/questions/45077190/airflow-dag-task-not-found-after-updating-adding-a-task

**Reply by potiuk** [reply]
I think somehow you get it into the situation that parsing the same DAG can lead to havin/not having that task, depending on circumstances of parsing (like time). Hard do say what causes it but yes the DAGs are parsed and re-parsed continuously (every 30 seconds by default) in order to make sure that nothing changed as a result of re-parse. The results are stored in SerializedDag table and webserver reads the structure of DAG from there. So if reparsing the DAG file can create different results, this might be the cause.

I think thought there were a few cases where stabilty of DAG serialization has been improved after 2.0.2. Hard to recall all the fixes but if you look at the changelog https://github.com/apache/airflow/blob/main/CHANGELOG.txt - there were quite a few fixes regarding serialization since 2.0.2, so I'd recommend to migrate to latest released versin and see if the problem persists.

**Reply by MaksymSofer** [reply]
Thank you for your prompt responses.
We will go with upgrade to the latest stable version
